### DISCUSSION TITLE
Workaround hiding HUD (Big Sur)

### DISCUSSION THREAD

**Discussion by DanielVolz** [AlexPerathoner/SlimHUD#23] [Show and tell]
This is not a real issue but just a workaround for "hiding" the macOS default HUD for changing brightness and volume since I'm not brave enough for the method mentioned here  https://github.com/AlexPerathoner/SlimHUD/issues/22#issuecomment-731397512.

This method actually doesn't hide the HUD but uses other tools and keys to trigger volume/brightness changes. 

**The cons of this method:**
- You have to use the f-keys as standard input and not the "picture-keys"
- You have to install two small programs 
- I don't know how this behaves with external monitors
- Pressing and holding a f-key to change brightness/volume will mess up stuff. Single taps are the way!

**The pros:**
- You don't see the ugly big HUD but the cool SlimHUD 😜😎

**Prerequisite:**
- You only need this if you are on Big Sur obviously
- Homebrew
- xcode command-line tools

**How to:**

1. Compile the brightness command line tool (https://github.com/nriley/brightness).
It is important to compile the programm from the github source and not use homebrew. The homebrew source is old and not maintained.
2. Install skhd following it's Github readme (https://github.com/koekeishiya/skhd).

3. Add these lines to the default skhd configurations file:
(If you don't have ggrep install it with: brew install grep)
```
# brightness down by 0.05
f1 : VAR1=$(brightness -lv | ggrep -oP '(?<=brightness ).*'); VAR2=$(echo  "$VAR1 - 0.05" | bc); brightness -m $VAR2
# brightness up by 0.05
f2 : VAR1=$(brightness -lv | ggrep -oP '(?<=brightness ).*'); VAR2=$(echo  "$VAR1 + 0.05" | bc); brightness -m $VAR2

# volume down by 7%
f11 : osascript -e "set volume output volume (output volume of (get volume settings) - 7) --100%"
# volume up by 7%
f12 : osascript -e "set volume output volume (output volume of (get volume settings) + 7) --100%"
```

4. "brew services restart skhd" to apply the changes in the skhd config.

**How it works**
The brightness program is used to read and change the current brightness levels via shell. Apples osascript is used to modify the volume levels. Skhd is a powerful hot key daemon which executes the above shell commands to change the volume/brightness levels on key press. 

***Update***

You can mute/unmute with this addition to the skhd config:

`f10 : flag=$(osascript -e 'output muted of (get volume settings)'); if [ "$flag" = true ]; then osascript -e 'set volume output muted false'; echo "laut"; else osascript -e 'set volume output muted true'; echo "stumm";fi`

**Comment by pieters2** [comment]
For hiding the hud in Big Sur I used this: 
https://apple.stackexchange.com/questions/16849/how-do-i-disable-the-volume-control-overlay 
I used the lines for Sierra, it worked really well! 



**To turn off bezels for this user until next login:**
launchctl unload -F /System/Library/LaunchAgents/com.apple.BezelUI.plist
To undo, change unload to load, or just log out and back in.

**For macOS 10.12 Sierra: Disable System Integrity protection, then:**
launchctl unload -F /System/Library/LaunchAgents/com.apple.OSDUIHelper.plist
Don't forget to enable System Integrity protection when you're done.

**To turn off bezels indefinitely for this user:**
launchctl unload -wF /System/Library/LaunchAgents/com.apple.BezelUI.plist

**For macOS 10.12 Sierra: Disable System Integrity protection, then:**
launchctl unload -wF /System/Library/LaunchAgents/com.apple.OSDUIHelper.plist
You can now enable System Integrity protection again -- your settings will persist.

To undo, change unload to load.

**To turn off bezels indefinitely for all users:**
sudo defaults write /System/Library/LaunchAgents/com.apple.BezelUI Disabled -bool YES

For macOS 10.11 Sierra: Disable System Integrity protection before doing the above.

**To undo, change YES to NO, or:**
sudo defaults delete /System/Library/LaunchAgents/com.apple.BezelUI Disabled

**For macOS 10.12 Sierra: Disable System Integrity protection, then:**
sudo defaults write /System/Library/LaunchAgents/com.apple.OSDUIHelper Disabled -bool YES
You can now enable System Integrity protection again -- your settings will persist.

**To undo, change YES to NO, or:**
sudo defaults delete /System/Library/LaunchAgents/com.apple.OSDUIHelper Disabled

Users can override this global setting using the two methods above.

**Reply by AlexPerathoner** [reply]
Unfortunately it doesn't seem to be working for me... changes take effect but are gone, as soon as I enable SIP and reboot the Mac.

**Reply by gaetgu** [reply]
This works great for me on Ventura keeping SIP disabled.

**Comment by AlexPerathoner** [comment]
Small update:
I just released [v1.4.0](https://github.com/AlexPerathoner/SlimHUD/releases/download/v1.4.0/SlimHUD.zip), which correctly implements this method. You can now use SlimHUD as soon as you launch it, without changing anything in the system :)

Note: due to changes in my signing certificates, this update can't be installed automatically so you'll have to download it manually
